drafts/multiformats.txt

```diff
@@ -0,0 +1,1008 @@
+
+Internet-Draft                Multiformats                 November 2019
+
+
+Table of Contents
+
+   1.  Introduction  . . . . . . . . . . . . . . . . . . . . . . . .   2
+   2.  Unsigned Variable Length Integer  . . . . . . . . . . . . . .   3
+     2.1.  ABNF Definition . . . . . . . . . . . . . . . . . . . . .   3
+   3.  Multibase . . . . . . . . . . . . . . . . . . . . . . . . . .   4
+     3.1.  Examples  . . . . . . . . . . . . . . . . . . . . . . . .   4
+     3.2.  Multibase ABNF  . . . . . . . . . . . . . . . . . . . . .   4
+   4.  Multihash . . . . . . . . . . . . . . . . . . . . . . . . . .   4
+     4.1.  Example Multihashes . . . . . . . . . . . . . . . . . . .   5
+     4.2.  Multihash ABNF  . . . . . . . . . . . . . . . . . . . . .   5
+   5.  Security Considerations . . . . . . . . . . . . . . . . . . .   5
+   6.  IANA Considerations . . . . . . . . . . . . . . . . . . . . .   5
+     6.1.  The Codec Registry  . . . . . . . . . . . . . . . . . . .   6
+       6.1.1.  Codec Registration  . . . . . . . . . . . . . . . . .   6
+       6.1.2.  Initial Contents  . . . . . . . . . . . . . . . . . .   7
+     6.2.  The Multibase Registry  . . . . . . . . . . . . . . . . .  15
+       6.2.1.  Base-Encoding Registration  . . . . . . . . . . . . .  15
+       6.2.2.  Initial Contents  . . . . . . . . . . . . . . . . . .  16
+   7.  Normative References  . . . . . . . . . . . . . . . . . . . .  17
+   Author's Address  . . . . . . . . . . . . . . . . . . . . . . . .  18
+
+1.  Introduction
+
+   Multiformats are a collection of data formats that aim to future-
+   proof systems by adding self-description to format values; allowing
+   interoperability and protocol agility while avoiding lock in.
+
+   The self-describing aspects of the Multiformat values have a few
+   stipulations:
+
+   o  They MUST be in-band (encoded within and carried with the value).
+
+   o  They MUST avoid lock-in and promote extensibility.
+
+   o  They MUST be compact and have a binary-packed representation.
+
+   o  They MUST have a human-readable representation.
+
+   This specification defines the Multihash and Multibase formats.
+
+   This specification also defines extensible registries for codec and
+   base-encoding identifiers.
+
+   New Multiformat data formats SHOULD be registered within the Codec
+   Registry using the "multiformat" type identifier.
+
+
+
+
+
+Snell                     Expires May 29, 2020                  [Page 2]
+
+Internet-Draft                Multiformats                 November 2019
+
+
+   In this document, the key words "MUST", "MUST NOT", "REQUIRED",
+   "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY",
+   and "OPTIONAL" are to be interpreted as described in [RFC2119].
+
+   This specification uses the Augmented Backus-Naur Form (ABNF)
+   notation of [RFC5234]
+
+2.  Unsigned Variable Length Integer
+
+   The various Multiformats defined in this specification make use of
```

Comment:
Could we mention that it is more generally spoken unsigned LEB128? It is used more broadly than just DWARF (https://en.wikipedia.org/wiki/LEB128#Uses).

Reply:
Is there a more normative URL reference for LEB128? Wikipedia really shouldn't be used for normative or informative links in RFCs

Reply:
I don't think we need to link to Wikipedia from the spec, this was just for your information. Also the DWARF spec mentions LEB128. I just would like to have "LEB128" mentioned in this spec so people don't need to pull up the DWARF spec, but can just see "oh, they are using LEB128".

One more note, could you please add that they need to be minimally encoded, that was recently added to the varint spec we currently have (https://github.com/multiformats/unsigned-varint/commit/1907e6a078cca5572a2140d1e9f713188f93ab06)